### PR TITLE
More minor code for nodegroup elements

### PR DESCRIPTION
**Pull request by nilsdeppe**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/DiscontinuousGalerkin/InboxTags.hpp [hunk 1]
```diff
@@ -236,6 +237,19 @@ struct BoundaryCorrectionAndGhostCellsInbox {
   void pup(PUP::er& /*p*/) {}
 };
 
+/*!
+ * \brief Simple tag used to store inbox data in the DataBox.
+ *
+ * Since the inbox data can be received asynchronously and lockfree ordered
+ * containers are slow and challenging to implement, we instead using an
+ * unordered container for the inbox, then transfer the data into an order map
```
Comment (kidder) [inline src/Evolution/DiscontinuousGalerkin/InboxTags.hpp]: ordered

**Review by kidder — CHANGES_REQUESTED** [review]
you can squash

File: src/Evolution/DiscontinuousGalerkin/InboxTags.hpp [hunk 2]
```diff
@@ -109,19 +113,47 @@ struct BoundaryCorrectionAndGhostCellsInbox {
...
+
+  // The actual type being used.
+  using type = type_map;
 
+  template <typename ReceiveDataType>
+  static size_t insert_into_inbox(const gsl::not_null<type_spsc*> inbox,
+                                  const temporal_id& time_step_id,
+                                  ReceiveDataType&& data) {
+    const DirectionalId<Dim>& neighbor_id = data.first;
+    // Note: This assumes the neighbor_id is oriented into our (the elment
```
Comment (knelli2) [inline src/Evolution/DiscontinuousGalerkin/InboxTags.hpp]: [nitpick] `elment` -> `element`

File: src/Evolution/DiscontinuousGalerkin/InboxTags.hpp [hunk 3]
```diff
@@ -109,19 +113,47 @@ struct BoundaryCorrectionAndGhostCellsInbox {
...
-        data.second;
-    (void)ghost_cell_data;
+  // Used by nodegroup implementation
+  using type_spsc = evolution::dg::AtomicInboxBoundaryData<Dim>;
+
+  // The actual type being used.
+  using type = type_map;
 
+  template <typename ReceiveDataType>
+  static size_t insert_into_inbox(const gsl::not_null<type_spsc*> inbox,
```
Comment (knelli2) [inline src/Evolution/DiscontinuousGalerkin/InboxTags.hpp:139]: You've changed the return type of this function. Could you add some docs about what the `size_t` represents? From the looks of it, it's just the new size of the inbox

File: src/Evolution/DiscontinuousGalerkin/InboxTags.hpp (hunk 1 above)
Comment (knelli2) [reply src/Evolution/DiscontinuousGalerkin/InboxTags.hpp]: [nitpick] also `using` -> `use`

**Review by knelli2 — CHANGES_REQUESTED** [review]

**Comment by nilsdeppe** [comment]
I've rebased and pushed fixups. I added a decent number of words for the return type being `size_t` since that is important. I'll need to clean this class up a bit after everything is working, mostly just the documentation. I also added `value_type` as a type alias since this is actually what the STL uses. It's not used right now, though.

**Review by kidder — COMMENTED** [review]
fixups LGTM

**Review by knelli2 — APPROVED** [review]
Fixups look good. Squash and rebase

**Comment by nilsdeppe** [comment]
Done! Thanks!

**Review by kidder — APPROVED** [review]

**Pull request merged**